test/docs/jdk/javadoc/doccheck/doccheckutils/checkers/ExtLinkChecker.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package doccheckutils.checkers;
+
+
+import doccheckutils.HtmlChecker;
+import doccheckutils.Log;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.MessageFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Checks the external links referenced in HTML files.
+ */
+public class ExtLinkChecker implements HtmlChecker, AutoCloseable {
+    private static final Path testBasePath = Path.of(System.getProperty("test.src"));
+    private static final Set<String> extLinks = new HashSet<>();
+
+    static {
+        String input = null;
+        try {
+            input = Files.readString(testBasePath.getParent().resolve("ExtLinksJdk.txt"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        extLinks.addAll(input.lines()
+                .map(line -> line.replaceAll("\\@\\@JAVASE_VERSION\\@\\@", String.valueOf(Runtime.version().feature())))
+                .filter(line -> !line.startsWith("#"))
```

Comment:
Nitpicking for mostly aesthetic reasons, but it would be nice to do the filter before the map, and pull the `String.valueof(...)` out into a variable.